test/fetch/http2.js

```diff
@@ -172,27 +171,29 @@ test('[Fetch] Should handle h2 request with body (string or buffer)', async t =>
 
   const responseBody = await response.text()
 
-  t.teardown(server.close.bind(server))
-  t.teardown(client.close.bind(client))
+  t.after(server.close.bind(server))
+  t.after(() => client.close())
```

Comment:
When I write this as
```js
t.after(client.close.bind(client))
```
it fails with the following error
```
Error [InvalidArgumentError]: invalid callback
      at Client.close (/home/anurag/dev/hobby/undici/lib/dispatcher-base.js:62:13)
      at TestHook.runInAsyncScope (node:async_hooks:206:9)
      at TestHook.run (node:internal/test_runner/test:631:25)
      at TestHook.run (node:internal/test_runner/test:856:18)
      at TestHook.run (node:internal/util:531:12)
      at node:internal/test_runner/test:565:20
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Test.runHook (node:internal/test_runner/test:563:7)
      at async after (node:internal/test_runner/test:591:9)
      at async Test.run (node:internal/test_runner/test:641:7) {
    code: 'UND_ERR_INVALID_ARG'
  }
```

On digging a little I found that an empty `TestContext {}` is being passed as callback to the `close` method. Not too sure why that is happening.